### PR TITLE
Changed UsageEventUtils to UsageEventEmitter and made it injectable

### PR DESCRIPTION
**Pull request by nnesic**
Refactored code to make the UsageEventEmitter injectable. Fixed unit tests where required, but otherwise no changes in the logic of event emitting.

### CODE REVIEW

File: engine/components-api/src/com/cloud/event/UsageEventEmitterImpl.java [hunk 1]
```diff
@@ -38,44 +34,37 @@
...
-    private static DataCenterDao s_dcDao;
-    private static final Logger s_logger = Logger.getLogger(UsageEventUtils.class);
-    protected static EventBus s_eventBus = null;
-    protected static ConfigurationDao s_configDao;
+public class UsageEventEmitterImpl implements UsageEventEmitter {
 
+    private static final Logger s_logger = Logger.getLogger(UsageEventEmitterImpl.class);
+
+    @Inject
+    UsageEventDao _usageEventDao;
```
Comment (DaanHoogland) [inline engine/components-api/src/com/cloud/event/UsageEventEmitterImpl.java]: please do not prefix members with '_'

File: engine/components-api/src/com/cloud/event/UsageEventEmitterImpl.java [hunk 2]
```diff
@@ -131,54 +127,55 @@ public static void publishUsageEvent(String usageType, long accountId, long zone
 
     }
 
-    private static void saveUsageEvent(String usageType, long accountId, long zoneId, long resourceId, String resourceName, Long offeringId, Long templateId,
+    private void saveUsageEvent(String usageType, long accountId, long zoneId, long resourceId, String resourceName, Long offeringId, Long templateId,
         String resourceType, Map<String, String> details) {
         UsageEventVO usageEvent = new UsageEventVO(usageType, accountId, zoneId, resourceId, resourceName, offeringId, templateId, resourceType);
-        s_usageEventDao.persist(usageEvent);
-        s_usageEventDao.saveDetails(usageEvent.getId(), details);
+        _usageEventDao.persist(usageEvent);
```
Comment (DaanHoogland) [inline engine/components-api/src/com/cloud/event/UsageEventEmitterImpl.java]: please change to usageEventDao. no reason in morden days to use these prefixes.

File: server/test/org/apache/cloudstack/networkoffering/MockUsageEventEmitter.java [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+import com.cloud.event.UsageEventEmitter;
+
+public class MockUsageEventEmitter implements UsageEventEmitter {
+
+    @Override
+    public void publishUsageEvent(String paramString1, long paramLong1,
+            long paramLong2, long paramLong3, String paramString2,
+            Long paramLong4, Long paramLong5, Long paramLong6,
+            String paramString3, String paramString4) {
+        // TODO Auto-generated method stub
```
Comment (DaanHoogland) [inline server/test/org/apache/cloudstack/networkoffering/MockUsageEventEmitter.java]: this (type of) comment can be removed

**Comment by ProjectMoon** [comment]
We need to re-submit this on master.

**Pull request closed without merging**